submissions/zalevskyi/friends-app/app.js

```diff
@@ -0,0 +1,133 @@
+const USERS_DATA = []
+const USERS_QUANTITY = 30
+const RESULTS = document.getElementById('results')
+const SORT = document.getElementById('sort')
+const FILTER = document.getElementById('filter')
+const REQUEST_PARAMETERS = {order: 'ascending'}
+let connectionAttemps = 1
+
+function connectToServer() {
+  RESULTS.innerHTML = `<div class='connection_message'>Please wait. Connecting to server. Attemp: ${connectionAttemps}</div>`
+  connectionAttemps++
+  fetchUserData(showResults, connectToServer)
+}
+function fetchUserData(onReady, onError) {
+  let apiUrl = 'https://randomuser.me/api/1.3/'
+  let fields = ['name', 'gender', 'location', 'dob', 'phone', 'picture']
+  fetch(`${apiUrl}?results=${USERS_QUANTITY}&inc=${fields.join(',')}`, {headers: {'Content-Type': 'application/json'}})
+    .then(response => {
+      if (!response.ok) {
+        throw Error(response.statusText)}
+      return response})
+    .then(response => response.json())
+    .then(responseJSON => {
+      if (responseJSON.hasOwnProperty('results')) {
+        pushUserData(responseJSON.results)
+        onReady()}
```

Comment:
No need for this `onReady`. You can return result of `fetch`, which is promise and then use `then` on it.